containers/ecr-viewer/src/app/utils/evaluate/fhir-paths.ts

```diff
@@ -0,0 +1,558 @@
+  patientSexualOrientation: ValueX;
+  patientGenderIdentity: ValueX;
+  patientReligion: ValueX;
+  patientMaritalStatus: ValueX;
+  eicrIdentifier: string;
+  eicrReleaseVersion: ValueX;
+  eicrCustodianRef: string;
+  dateTimeEcrCreated: string;
+  ehrSoftware: ValueX;
+  ehrManufacturerModel: string;
+  eRSDwarnings: Coding;
+  compositionAuthorRefs: Reference;
+  encounterEndDate: string;
+  encounterStartDate: string;
+  encounterDiagnosis: EncounterDiagnosis;
+  encounterType: string;
+  encounterID: Identifier;
+  facilityContact: string;
+  facilityContactAddress: string;
+  facilityLocation: string;
+  facilityName: string;
+  facilityAddress: Address;
+  facilityType: ValueX;
+  compositionEncounterRef: string;
+  encounterIndividualRef: string;
+  encounterParticipants: EncounterParticipant;
+  rrDetails: Observation;
+  clinicalReasonForVisit: ValueX;
+  patientHeight: ValueX;
+  patientHeightDate: string;
+  patientWeight: ValueX;
+  patientWeightDate: string;
+  patientBmi: ValueX;
+  patientBmiDate: string;
+  resolve: unknown;
+  activeProblems: Condition;
+  activeProblemsDisplay: string;
+  activeProblemsOnsetDate: string;
+  activeProblemsOnsetAge: ValueX;
+  activeProblemsComments: string;
+  historyOfPresentIllness: string;
+  planOfTreatment: string;
+  plannedProcedures: CarePlanActivity;
+  plannedProcedureName: string;
+  plannedProcedureOrderedDate: string;
+  plannedProcedureScheduledDate: string;
+  adminMedicationsRefs: string;
+  careTeamParticipants: CareTeamParticipant;
+  careTeamParticipantMemberName: string;
+  careTeamParticipantRole: string;
+  careTeamParticipantStatus: string;
+  careTeamParticipantPeriod: string;
+  immunizations: Immunization;
+  immunizationsName: string;
+  immunizationsAdminDate: string;
+  immunizationsDoseNumber: ValueX;
+  immunizationsManufacturerName: string;
+  immunizationsLotNumber: unknown;
+  procedures: Procedure;
+  procedureName: string;
+  procedureDate: string;
+  procedureReason: string;
+  diagnosticReports: DiagnosticReport;
+  observations: Observation;
+  labResultDiv: string;
+  specimenCollectionTime: string;
+  specimenReceivedTime: string;
+  specimenSource: string;
+  observationReferenceValue: string;
+  observationComponent: string;
+  observationValue: string;
+  observationReferenceRange: string;
+  observationDeviceReference: Reference;
+  observationNote: string;
+  observationOrganism: string;
+  observationAntibiotic: string;
+  observationOrganismMethod: string;
+  observationSusceptibility: string;
+  organizations: Organization;
+  patientTravelHistory: Observation;
+  travelHistoryStartDate: string;
+  travelHistoryEndDate: string;
+  travelHistoryLocation: string;
+  travelHistoryPurpose: string;
+  stampedImmunizations: Immunization;
+};
+
+type PathKeys = keyof PathTypes;
+
+export interface FhirPath<K> {
+  type: string;
+  path: string;
+  name: K;
+}
+
+const _fhirPathMappings: { [K in PathKeys]: Omit<FhirPath<K>, "name"> } = {
```

Comment:
TODO: think about if we can add a value/type here that says whether we expect one or many and then use that to dispatch appropriately

Reply:
after chatting with @jakewheeler and @JNygaard-Skylight , we decided this wouldn't be an improvement